koitoror/SendIT
# Ft user login 161739482

#### What does this PR do?
Creates API endpoint to : 
- Login a user

#### Description of Task to be completed?
Have the `users.py` working [with login views update]
Have the `decorators.py` working
Have the `validators.py` working


#### How should this be manually tested?
For the Tests to work you need to do the following steps

Clone the repository into your local environment

```
$ git clone https://github.com/koitoror/SendIT.git
```

Switch to the USER-LOGIN branch
```
$ git checkout ft-user-login-161739482
```

Navigate to api folder.
```
cd SendIT
```

Create a virtual environment.
Use [virtualenv](http://www.pythonforbeginners.com/basics/how-to-use-python-virtualenv) for an isolated working environment.

```
virtualenv venv --python=python3
```

Install the packages.
```
pip3 install -r requirements.txt
```

Set environment variables for 

> `SECRET_KEY` is your secret key

> `FLASK_CONFIG` is the environment you are running on. Should be either `Production`, `Development` or `Testing`. NOTE: its case sensitive

To get the app running...

```bash
$ python run.py run
```

Open the root path in your browser to test the endpoints. 
You can also use Postman or any other agent to test the endpoints

On Postman enter the relevant URL which is 
>http://127.0.0.1:5000/api/v1/auth/login
for logging in a user. 
If it returns a "message": "Logged in successfully" and a **token** string generated then we have successfully logged in a user.


#### Any background context you want to provide?


#### What are the relevant pivotal tracker stories?
https://www.pivotaltracker.com/n/projects/2216466
#161739480

#### Screenshots (if appropriate)


-
